cli/download_source_test.go

```diff
@@ -21,6 +21,24 @@ import (
 	"github.com/gruntwork-io/terragrunt/util"
 )
 
+func TestAlreadyHaveLatestCodeLocalFilePathWithHash(t *testing.T) {
+	t.Parallel()
+
+	canonicalUrl := fmt.Sprintf("file://%s", absPath(t, "../test/fixture-download-source/hello-world-local"))
+	downloadDir := fmt.Sprintf("%s", absPath(t, "./test-fixtures/download-dir-local-no-changes"))
+
+	testAlreadyHaveLatestCode(t, canonicalUrl, downloadDir, true)
+}
+
+func TestAlreadyHaveLatestCodeLocalFilePathWithHashModified(t *testing.T) {
+	t.Parallel()
+
+	canonicalUrl := fmt.Sprintf("file://%s", absPath(t, "../test/fixture-download-source/hello-world-local"))
+	downloadDir := fmt.Sprintf("%s", absPath(t, "./test-fixtures/download-dir-local-modified"))
```

Comment:
Any reason we can't make this download dir a tmpDir? Does it need to persist in the codebase? (same comment)

Reply:
Also another small nit. The `cli_app_test.go` uses the `cli/test-fixtures` directory, but none of the tests in `download_source_test.go` use it. Would it be fine to continue using the downloadDirs we use elsewhere in this file? E.g.:

```go
downloadDir := "../test/fixture-download-source/download-dir-XXX"
```

Reply:
Refactored the code as you suggested.
Remove most of the files.
Now have 2 folders like the most other tests to be able to compare `tf` contents hash with the `version-file.txt`.
Or would you like to generate the files in the test case?